pkg/datastore/mysql/query.go

```diff
@@ -93,3 +98,45 @@ func toMySQLDirection(d datastore.OrderDirection) string {
 		return ""
 	}
 }
+
+func refineOrdersField(orders []datastore.Order) []datastore.Order {
+	for i, order := range orders {
+		switch order.Field {
+		case "SyncState.Status":
+			orders[i].Field = "SyncState_Status"
+		default:
+			continue
+		}
+	}
+	return orders
+}
+
+func refineFiltersField(filters []datastore.ListFilter) []datastore.ListFilter {
+	for i, filter := range filters {
+		switch filter.Field {
+		case "SyncState.Status":
+			filters[i].Field = "SyncState_Status"
+		default:
+			continue
+		}
+	}
+	return filters
+}
+
+func refineFiltersOperator(filters []datastore.ListFilter) ([]datastore.ListFilter, error) {
+	for i, filter := range filters {
+		switch filter.Operator {
```

Comment:
It is ok for now. But in the future we might define a type for Operator instead of directly using string. Then we have a list of usable operators. Inside each backend package(firestore, mysql) we convert them to the appropriate string.

Reply:
Nice idea, will create an issue for that and address it later 👍